### PR TITLE
Airlock name mapping helper

### PR DESCRIPTION
**Pull request by unit0016**
## About The Pull Request
The moment I saw #65580 go up, I lit up like a lightbulb. A _very deranged_ lightbulb. This PR introduces a new mapping helper to help cut down on map varedits - the autoname helper.
It finds the `/area/` it's in, and automatically renames the door it's placed on the same.
image

~~I did run into a dread BYOND moment while writing this up but thankfully it was easy enough to work around.~~

## Why It's Good For The Game
Makes it easier both to visualize the dominant `/area/` in-game (Doors X and Y are hooked to Z APC) - and to.. well, name airlocks as a mapper! It's the little things, y'know?

## Changelog

:cl:
qol: Mappers now have access to an autoname helper for airlocks! One less field to punch in means a few seconds extra YOU can use placing down lizard plushies, wondering where it all went wrong, and adding lattices on the CentCom Z-level.
/:cl:

### CODE REVIEW

**Comment by dragomagol** [comment]
🤝

**Comment by cacogen** [comment]
I would've done this in the Initialize() of airlocks, personally, to avoid even having to use a helper. Check for mapload, check if the airlock (or maybe all doors) has the initial name (e.g. "airlock") and if so rename it based on the area it's in.

**Comment by unit0016** [comment]
> I would've done this in the Initialize() of airlocks, personally, to avoid even having to use a helper. Check for mapload, check if the airlock (or maybe all doors) has the initial name (e.g. "airlock") and if so rename it based on the area it's in.

I was considering flipping it around that way for a bit after I made this PR and converting the helper into a blacklister, but've figured it's probably fine as is.

**Review by Ryll-Ryll — APPROVED** [review]
Fix the merge conflict and we good

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -204,6 +205,8 @@
 				panel_open = TRUE
 	if(cutAiWire)
 		wires.cut(WIRE_AI)
+	if(autoname)
+		name = "\proper [get_area_name(src)]" //Formatted this way to work around a BYOND issue : "That's a Primary Tool Storage"
```
Comment (ShizCalev) [inline code/game/machinery/doors/airlock.dm:209]: name = get_area_name(src, TRUE)

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (unit0016) [reply code/game/machinery/doors/airlock.dm]: Added in commit be04afa

**Pull request merged**